githooks/pre-commit

```diff
@@ -1,15 +1,19 @@
 #!/bin/bash
-set -evx
-for f in ./*modules*
-do
+
+for f in ./*modules*; do
     cd "$f"
-    for i in `ls -d */` do 
-        git diff
-        terraform-docs markdown table --output-file README.md --output-mode inject "${i}";
+    for i in `ls -d */`;do
+        if ! git diff --quiet "${i}" 
+        then
+            terraform-docs markdown table --output-file README.md --output-mode inject "${i}";
+        fi
     done
-    for y in $(find . -name '*.tf' -print) do
-        git diff
-        terraform fmt -write=true "${y}"
+    for y in $(find . -name '*.tf' -print)
```

Comment:
it seems we can move this block outside of for containing for block,
also have some indicator of start end and in progress processing, 
something like 
echo "Start terraform fmt"; # at beginning of terraform fmt and 
echo "End terraform fmt"; # at end , and 
echo "Formatting the file: ${y}"; # in the place just before processing a file

similar one for terraform-docs

Reply:
I change the script. Please check is I understand right the problem?